### DISCUSSION TITLE
[Feature] Proper image selection.

### DISCUSSION THREAD

**Discussion by Dental1347** [immich-app/immich#2281] [Feature Request]
### The feature

Hello,
Within this post I hope to outline a view requests I would like to be added within the web-UI of Immich. I have checked the feature requests, however I can't say for certain there isn't duplication with other posts.
1. Currently, there is no way to select all the photos from within the web-UI. This could be achieved by implementing a 'select all' or allow Ctrl + a to easily select all images.
2. The ability to select the first image; then hold shift and select the next image with left mouse-click. Currently this is not possible, and when trying it will open the next you click (even when clicking on the check) 
![image](https://user-images.githubusercontent.com/44779473/232823012-73013aa3-2c02-4ffb-bd8f-b6509143101b.png)
This lack of support becomes especially apparent when going to explore, where you will have to select all pictures individually.

For mobile phone the first option still applies, however it would best work with a select all button.
At the same time on mobile devices, you need to select on a per-image basis or select the group by date. This can be made smoother and more intuitive. A possibility here could be to select and hold 1 image and drag down to select multiple images, this would make the image selection way more intuitive and user friendly. At the same time, and I know this is in the works, when uploading images they don't automatically get uploaded in an album but just straight to the time line. When you manually want to create an album, there is a lot of image selections you will need to perform.

When more requets related to image selection come up I will list them here:
---

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1659

**Reply by Dental1347** [reply]
Thanks for pointing that out @bo0tzz.

**Comment by Dental1347** [comment]
I want to add something to this post:

When selecting an image group, as per shown below, it will then select the whole group. 
![image](https://user-images.githubusercontent.com/44779473/232820783-e3d81fca-d7f4-498c-8274-3839b2f08638.png)
When holding shift and clicking on the check in the top left corner of the image, it will rapidly select and deselect the image, for as long as you have shift pressed, you can even let go of the mouse as it will still do this. When you click the image itself it does not do this.

This was discovered when trying to click an image and then hold shift and select another image. 
Illustration:
https://www.youtube.com/watch?v=6YYhgpf8wJY
